### PR TITLE
Add support to NUMBER to Python UDFs

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
## Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java [hunk 1]
```diff
@@ -201,6 +204,18 @@ private static void javaToBinary(Type type, Object value, SliceOutput output)
                         : Decimals.toString((Int128) value, decimalType.getScale());
                 writeVariableSlice(utf8Slice(decimalString), output);
             }
+            case NumberType _ -> {
+                TrinoNumber number = (TrinoNumber) value;
+
+                switch (number.toBigDecimal()) {
+                    case TrinoNumber.BigDecimalValue bigDecimalValue ->
```
Comment (findepi) [inline plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java]:
```suggestion
                    case TrinoNumber.BigDecimalValue(BigDecimal  bigDecimal) ->
```

File: plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java [hunk 2]
```diff
@@ -201,6 +204,18 @@ private static void javaToBinary(Type type, Object value, SliceOutput output)
...
+                            writeVariableSlice(utf8Slice(bigDecimalValue.toString()), output);
+                    case TrinoNumber.Infinity infinity ->
```
Comment (findepi) [inline plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java]: You can pull `writeVariableSlice(utf8Slice(` outside of `switch`

File: plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java [hunk 3]
```diff
@@ -201,6 +204,18 @@ private static void javaToBinary(Type type, Object value, SliceOutput output)
...
+                            writeVariableSlice(infinity.negative() ? utf8Slice("-Infinity") : utf8Slice("Infinity"), output);
+                    case TrinoNumber.NotANumber _ ->
```
Comment (findepi) [inline plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java]:
```suggestion
                    case TrinoNumber.NotANumber() ->
```

File: plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java (hunk 2 above)
Comment (findepi) [inline plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java]:
```suggestion
                    case TrinoNumber.Infinity(boolean negative) ->
```

File: plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java [hunk 4]
```diff
@@ -156,7 +159,7 @@ private static TrinoType singletonType(Type type)
             case StandardTypes.TINYINT -> TrinoType.TINYINT;
             case StandardTypes.DOUBLE -> TrinoType.DOUBLE;
             case StandardTypes.REAL -> TrinoType.REAL;
-            case StandardTypes.DECIMAL -> TrinoType.DECIMAL;
+            case StandardTypes.DECIMAL, StandardTypes.NUMBER -> TrinoType.DECIMAL;
```
Comment (findepi) [inline plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java:169]: What is the semantics of TrinoType enum?
I don't see javadoc nor usages on it... Why is it safe to present NUMBER as a decimal?

File: plugin/trino-functions-python/src/test/java/io/trino/plugin/functions/python/TestPythonFunctions.java [hunk 5]
```diff
@@ -751,6 +751,27 @@ SELECT bad_bigint_return()
...
+                WITH (handler = 'multiply')
+                AS $$
+                from decimal import Decimal
+                def multiply(x, y):
+                    return x * y * Decimal("100000000000000000000.000000000000000000000000000000000000001")
+                $$
+                """;
+
+        assertThat(assertions.query(
+                query + "SELECT multiply(NUMBER '1.12345', NUMBER '2.54321')"))
```
Comment (findepi) [inline plugin/trino-functions-python/src/test/java/io/trino/plugin/functions/python/TestPythonFunctions.java:772]: Can you please add a test with NULL, NaN and ±Infinity?

File: plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java (hunk 4 above)
Comment (wendigo) [reply plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java:169]: It's a translation between the host and the wasm environment. Types are covered here: https://github.com/trinodb/trino-wasm-python/blob/master/pyhost.h#L10

File: plugin/trino-functions-python/src/test/java/io/trino/plugin/functions/python/TestPythonFunctions.java (hunk 5 above)
Comment (wendigo) [reply plugin/trino-functions-python/src/test/java/io/trino/plugin/functions/python/TestPythonFunctions.java:772]: NaN and +/- Infinity are not supported by wasm python yet so nothing to test.

File: plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java (hunk 4 above)
Comment (findepi) [reply plugin/trino-functions-python/src/main/java/io/trino/plugin/functions/python/TrinoTypes.java:169]: to me this looks like we need to expand that enum when a new trino type is added

File: plugin/trino-functions-python/src/test/java/io/trino/plugin/functions/python/TestPythonFunctions.java (hunk 5 above)
Comment (findepi) [reply plugin/trino-functions-python/src/test/java/io/trino/plugin/functions/python/TestPythonFunctions.java:772]: the behavior should still be tested, regardless of current behavior
